sdk/cosmosdbforpostgresql/azure-resourcemanager-cosmosdbforpostgresql/src/test/java/com/azure/resourcemanager/cosmosdbforpostgresql/CosmosDBForPostgreSqlManagerTests.java

```diff
@@ -0,0 +1,118 @@
+    private ResourceManager resourceManager;
+    private boolean testEnv;
+
+    @Override
+    public void beforeTest() {
+        final TokenCredential credential = new DefaultAzureCredentialBuilder().build();
+        final AzureProfile profile = new AzureProfile(AzureEnvironment.AZURE);
+
+        cosmosDBForPostgreSqlManager = CosmosDBForPostgreSqlManager
+            .configure()
+            .withLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC))
+            .authenticate(credential, profile);
+
+        resourceManager = ResourceManager
+            .configure()
+            .withLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC))
+            .authenticate(credential, profile)
+            .withDefaultSubscription();
+
+        // use AZURE_RESOURCE_GROUP_NAME if run in LIVE CI
+        String testResourceGroup = Configuration.getGlobalConfiguration().get("AZURE_RESOURCE_GROUP_NAME");
+        testEnv = !CoreUtils.isNullOrEmpty(testResourceGroup);
+        if (testEnv) {
+            resourceGroupName = testResourceGroup;
+        } else {
+            resourceManager.resourceGroups()
+                .define(resourceGroupName)
+                .withRegion(REGION)
+                .create();
+        }
+    }
+
+    @Override
+    protected void afterTest() {
+        if (!testEnv) {
+            resourceManager.resourceGroups().beginDeleteByName(resourceGroupName);
+        }
+    }
+
+    @Test
+    @DoNotRecord(skipInPlayback = true)
+    public void testCreateCluster() {
+        Cluster cluster = null;
+        String  randomPadding = randomPadding();
+        try {
+            String clusterName = "cluster" + randomPadding;
+            String adminPwd = "Pass@" + randomPadding;
+            // @embedmeStart
+            cluster = cosmosDBForPostgreSqlManager
+                .clusters()
+                .define(clusterName)
+                .withRegion(REGION)
+                .withExistingResourceGroup(resourceGroupName)
+                .withAdministratorLoginPassword(adminPwd)
+                .withPostgresqlVersion("15")
+                .withCitusVersion("12.1")
+                .withMaintenanceWindow(new MaintenanceWindow()
+                    .withCustomWindow("Disabled")
+                    .withDayOfWeek(0)
+                    .withStartHour(0)
+                    .withStartMinute(0))
+                .withEnableShardsOnCoordinator(true)
+                .withEnableHa(false)
+                .withCoordinatorServerEdition("GeneralPurpose")
+                .withNodeServerEdition("MemoryOptimized")
+                .withCoordinatorStorageQuotaInMb(131072)
+                .withNodeStorageQuotaInMb(524288)
+                .withCoordinatorVCores(2)
+                .withNodeVCores(4)
+                .withCoordinatorEnablePublicIpAccess(true)
+                .withNodeEnablePublicIpAccess(true)
+                .withNodeCount(0)
```

Comment:
What does nodeCount=0 mean?

Reply:
When node count is 0, it represents a single node configuration with the ability to create distributed tables on that node. 2 or more worker nodes represent multi-node configuration. Node count value cannot be 1. Required for creation

Reply:
Got it, thanks.